spring-cql/src/main/java/org/springframework/cassandra/config/PoolingOptionsFactoryBean.java

```diff
@@ -15,226 +15,455 @@
-	private Integer remoteMinSimultaneousRequests;
-	private Integer remoteMaxSimultaneousRequests;
+	private Integer localMaxSimultaneousRequests;
+	private Integer localMinSimultaneousRequests;
+	private Integer poolTimeoutMilliseconds;
 	private Integer remoteCoreConnections;
 	private Integer remoteMaxConnections;
+	private Integer remoteMaxSimultaneousRequests;
+	private Integer remoteMinSimultaneousRequests;
 
 	PoolingOptions poolingOptions;
 
-	@Override
-	public void destroy() throws Exception {
-		localMinSimultaneousRequests = null;
-		localMaxSimultaneousRequests = null;
-		localCoreConnections = null;
-		localMaxConnections = null;
-		remoteMinSimultaneousRequests = null;
-		remoteMaxSimultaneousRequests = null;
-		remoteCoreConnections = null;
-		remoteMaxConnections = null;
-	}
-
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.beans.factory.InitializingBean#afterPropertiesSet()
+	 */
 	@Override
 	public void afterPropertiesSet() throws Exception {
 
-		poolingOptions = new PoolingOptions();
+		poolingOptions = newPoolingOptions();
 
-		if (localMaxConnections != null) {
-			poolingOptions.setMaxConnectionsPerHost(HostDistance.LOCAL, localMaxConnections);
+		if (heartbeatIntervalSeconds != null) {
+			poolingOptions.setHeartbeatIntervalSeconds(heartbeatIntervalSeconds);
 		}
 
-		if (localCoreConnections != null) {
-			poolingOptions.setCoreConnectionsPerHost(HostDistance.LOCAL, localCoreConnections);
+		if (idleTimeoutSeconds != null) {
+			poolingOptions.setIdleTimeoutSeconds(idleTimeoutSeconds);
 		}
 
-		if (localMinSimultaneousRequests != null) {
-			/*
-			 * If the new min is greater than the current Max, set the current max to the new min first.
-			 * This is enforced by the DSE Driver so you cannot set a new min/max together if either one falls outside of the default 25-100 range.
-			 */
-			int currentMax = poolingOptions.getNewConnectionThreshold(HostDistance.LOCAL);
-			if (currentMax < localMinSimultaneousRequests) {
-				poolingOptions.setNewConnectionThreshold(HostDistance.LOCAL,
-						localMinSimultaneousRequests);
-			}
+		if (initializationExecutor != null) {
+			poolingOptions.setInitializationExecutor(initializationExecutor);
 		}
 
-		if (localMaxSimultaneousRequests != null) {
-			poolingOptions.setMaxRequestsPerConnection(HostDistance.LOCAL, localMaxSimultaneousRequests);
+		if (poolTimeoutMilliseconds != null) {
+			poolingOptions.setPoolTimeoutMillis(poolTimeoutMilliseconds);
 		}
 
-		if (remoteMaxConnections != null) {
-			poolingOptions.setMaxConnectionsPerHost(HostDistance.REMOTE, remoteMaxConnections);
-		}
+		HostDistancePoolingOptions local = LocalHostDistancePoolingOptions.create(localCoreConnections,
+			localMaxConnections, localMaxSimultaneousRequests, localMinSimultaneousRequests);
 
-		if (remoteCoreConnections != null) {
-			poolingOptions.setCoreConnectionsPerHost(HostDistance.REMOTE, remoteCoreConnections);
-		}
+		local.setMaxConnectionsPerHost(poolingOptions);
```

Comment:
Maybe add another method to extract the same calls for `HostDistancePoolingOptions`.